src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -1247,7 +1247,16 @@ protected override void ProcessRecord()
                         StopDependentService(process);
                     }
 
-                    if (!process.HasExited)
+                    // If the process has not exited and we are not Forcing, attempt close request 
+                    if (!process.HasExited && !Force)
+                    {
+                        _closeRequest = process.CloseMainWindow();
+                    }
+
+                    // If CloseMainWindow request was not successful, or unsupported, fallback to Kill
+                    // In some cases, this can have the affect that running Stop-Process without Force 
+                    // twice would revert to Kill()
+                    if (!_closeRequest)
                     {
                         process.Kill();
                     }
```

Comment:
I think we have no need to have _closeRequest as new field and we could simplify logic, make the code more readable and remove obvious comments - only comments are describing scenarios would be useful.

```c#
if (process.HasExited)
{
    return;
}

if (Force || !process.CloseMainWindow())
{
    process.Kill();
}
```